File: frontend/src/types/documents.ts
export type DocFileType = 'doc' | 'docx' | 'odt' | 'txt'

export interface DocItem {
  id: string
  filename: string
  display_name: string
  note: string
  source_path: string
  text_path: string
  char_count: number
  file_type: DocFileType
  file_size: number
  created_at: string
  updated_at: string
}

export interface AddDocumentRequest extends Record<string, unknown> {
  filePath: string
}

File: backend/app/schemas/documents.py
from typing import Literal

from pydantic import BaseModel


DocFileType = Literal["doc", "docx", "odt", "txt"]


class DocItem(BaseModel):
    id: str
    filename: str
    display_name: str
    note: str
    source_path: str
    text_path: str
    char_count: int
    file_type: DocFileType
    file_size: int
    created_at: str
    updated_at: str
